src/protocols/radius/encode.c

```diff
@@ -303,14 +319,16 @@ static ssize_t encode_tlv_hdr_internal(uint8_t *out, size_t outlen,
 		vp = fr_cursor_current(cursor);
 	}
 
-	return p - out;
+	return init_remaining - fr_dbuff_remaining(dbuff);
```

Comment:
This should just be `return fr_dbuff_used(dbuff)`

here, and other places.

Reply:
`fr_dbuff_used(dbuff)` includes bytes already written to dbuff, so just doing that doesn't work. What is needed is no-advance dbuffs having start set to the parent's p (or more accurately the same thing as their p is set to; the `FR_DBUFF_NO_ADVANCE_RESERVE()` macro has an edge case); actually, I'm not sure why that's not done for all children. Then you can write

```
fr_dbuff_t work_dbuff = FR_DBUFF_NO_ADVANCE(dbuff);
/* Do what's needed using work_dbuff */
return fr_dbuff_advance(dbuff, fr_dbuff_used(&work_dbuff));
```

Reply:
Yeah, that's the pattern which should be used everywhere.